gnovm/pkg/gnolang/precompile.go

```
@@ -161,18 +162,19 @@ func Precompile(source string, tags string, filename string) (*precompileResult,
 
 	transformed, err := precompileAST(fset, f, shouldCheckWhitelist)
 	if err != nil {
-		return nil, fmt.Errorf("%w", err)
+		return nil, fmt.Errorf("precompileAST: %w", err)
 	}
 
 	header := "// Code generated by github.com/gnolang/gno. DO NOT EDIT.\n\n"
 	if tags != "" {
 		header += "//go:build " + tags + "\n\n"
 	}
-	_, err = out.WriteString(header)
+	var out bytes.Buffer
+	out.WriteString(header)
+	err = format.Node(&out, fset, transformed)
 	if err != nil {
-		return nil, fmt.Errorf("write to buffer: %w", err)
+		return nil, fmt.Errorf("format.Node: %w", err)
 	}
-	err = format.Node(&out, fset, transformed)
```

can't believe this error was swallowed :facepalm: